### DISCUSSION TITLE
Documentation mismatch: "packageRule must contain at least one match"

### DISCUSSION THREAD

**Discussion by jamie-tm** [renovatebot/renovate#31509] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub, renovate-ce-ee 8.2.0

### Please tell us more about your question or problem

I believe there's a mismatch between the documentation and the actual behavior. Though, I'm not sure which one is intended to be correct.

The [documentation for `packageRules`](https://docs.renovatebot.com/configuration-options/#packagerules) states:
> If no match/exclude matchers are defined, everything matches.

However, the [validation](https://github.com/renovatebot/renovate/blob/b882965ef63e65a0c9966aaa94ddbbbfb4b844d3/lib/config/validation.ts#L471) fails if this is the case:
> Each packageRule must contain at least one match* or exclude* selector.

Either behavior seems perfectly reasonable, but one of them should be updated to reflect the other.

For context, my use-case was to override the `automerge` setting for one particular repo, while otherwise using an org-wide preset. Setting `automerge: false` in the root of the config didn't work, because it was overridden by inherited package rules. So I tried to make match-any package rule per the documentation, and got the error.

This is ultimately my solution, and seems to work:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>MY_ORG/renovate-config"
  ],
  "packageRules": [
    {
      "description": "Do not auto-merge in this repo",
      "matchPackageNames": ["*"],
      "automerge": false
    }
  ]
}
```

### Logs (if relevant)

<details><summary>Logs</summary>

n/a

</details>

**Comment by secustor** [comment]
Good catch. 

Are you interest in providing a PR to change the phrasing to something like the following?

```diff
- If no match/exclude matchers are defined, everything matches.
+ At least one matcher has to be included in a `packageRule`
```

**Comment by jamie-tm** [comment, marked as answer]
I'd be happy to. :)
https://github.com/renovatebot/renovate/pull/31512
